packages/overlay/test/restore-focus.test.js

```diff
@@ -58,9 +58,32 @@ describe('restore focus', () => {
       expect(isElementFocused(focusInput)).to.be.false;
     });
 
-    describe('restoreFocusOnClose', () => {
+    describe('restoreFocusNode', () => {
+      beforeEach(() => {
+        overlay.restoreFocusNode = focusInput;
+      });
+
+      it('should not restore focus on close by default', async () => {
+        focusInput.focus();
+        await open(overlay);
+        await close(overlay);
+        expect(isElementFocused(focusInput)).to.be.false;
+      });
+    });
+  });
+
+  describe('restoreFocusOnClose', () => {
+    beforeEach(() => {
+      overlay.restoreFocusOnClose = true;
+    });
+
+    describe('basic', () => {
```

Comment:
https://github.com/vaadin/web-components/pull/5839 will add two additional test suites "focus node inside a slot" and "focus node inside Shadow DOM" next to "basic".